code/modules/reagents/reagent_containers/dropper.dm

```diff
@@ -1,116 +1,111 @@
 ////////////////////////////////////////////////////////////////////////////////
-/// Droppers.
+/// 								Droppers.								 ///
 ////////////////////////////////////////////////////////////////////////////////
+
 /obj/item/weapon/reagent_containers/dropper
 	name = "dropper"
 	desc = "A dropper. Transfers 5 units."
-	icon = 'icons/obj/chemical.dmi'
 	icon_state = "dropper"
+	item_state = "dropper"
 	amount_per_transfer_from_this = 5
-	possible_transfer_amounts = list(1,2,3,4,5)
+	possible_transfer_amounts = list(1, 2, 3, 4, 5)
 	volume = 5
 
 /obj/item/weapon/reagent_containers/dropper/on_reagent_change()
-	if(reagents.total_volume <= 0)
+	if(!reagents.total_volume)
 		icon_state = "[initial(icon_state)]"
 	else
 		icon_state = "[initial(icon_state)]1"
 
-/obj/item/weapon/reagent_containers/dropper/afterattack(obj/target, mob/user , flag)
-	if(!target.reagents)
-		return
-
-	if(reagents.total_volume)
+/obj/item/weapon/reagent_containers/dropper/attack(mob/living/M, mob/living/user, def_zone)
+	return
 
-		if(target.reagents.total_volume >= target.reagents.maximum_volume)
-			to_chat(user, "<span class='warning'>[target] is full.</span>")
+/obj/item/weapon/reagent_containers/dropper/afterattack(atom/target, mob/user, proximity)
+	if(!proximity)
+		return
+	var/to_transfer = 0
+	if(iscarbon(target))
+		var/mob/living/carbon/C = target
+		if(!reagents.total_volume)
 			return
-
-		if(!target.is_open_container() && !ismob(target) && !istype(target,/obj/item/weapon/reagent_containers/food) && !istype(target, /obj/item/clothing/mask/cigarette)) //You can inject humans and food but you cant remove the shit.
-			to_chat(user, "<span class='warning'>You cannot directly fill this object.</span>")
+		if(user != C)
+			visible_message("<span class='danger'>[user] begins attempting to squirt something into [C]'s eyes!</span>")
+			if(!do_mob(user, C, 30))
+				return
+		if(ishuman(target))
+			var/mob/living/carbon/human/H = target
+			var/obj/item/safe_thing = null
+
+			if(H.glasses)
+				safe_thing = H.glasses
+			if(H.wear_mask )
+				if(H.wear_mask.flags_cover & MASKCOVERSEYES)
+					safe_thing = H.wear_mask
+			if(H.head)
+				if(H.head.flags_cover & MASKCOVERSEYES)
+					safe_thing = H.head
+
+			if(safe_thing)
+				visible_message("<span class='danger'>[user] tries to squirt something into [H]'s eyes, but fails!</span>")
+
+				reagents.reaction(safe_thing, TOUCH)
+				to_transfer = reagents.remove_any(amount_per_transfer_from_this)
+
+				to_chat(user, "<span class='notice'>You transfer [to_transfer] units of the solution.</span>")
+				return
+
+		visible_message("<span class='danger'>[user] squirts something into [C]'s eyes!</span>")
+		reagents.reaction(C, TOUCH)
+
+		var/list/injected = list()
+		for(var/datum/reagent/R in reagents.reagent_list)
+			injected += R.name
+		var/contained = english_list(injected)
+		add_logs(user, C, "squirted", src, "([contained])")

```

Comment:
Actually, this comes out to:

`[01:00:57] Has dripped Fox P McCloud/(Nicvuhgen Farierengvuh) with the dropper(Hydrogen, Copper and Bromine) = null`
 
`[01:00:57] Has been dripped by Fox P McCloud/(Nicvuhgen Farierengvuh) with the dropper(Hydrogen, Copper and Bromine) = null`

if I just change it to `dripped`.